src/main/java/ITNews/project/ITNews/service/CloudinaryService.java

```diff
@@ -0,0 +1,84 @@
+package ITNews.project.ITNews.service;
+
+import ITNews.project.ITNews.dto.ControllerResponse;
+import ITNews.project.ITNews.model.ImgNewsEntity;
+import ITNews.project.ITNews.repository.ImgNewsRepository;
+import com.cloudinary.Cloudinary;
+import com.cloudinary.utils.ObjectUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+@Service
+@Transactional
+public class CloudinaryService {
+    private static final String noImg= "https://res.cloudinary.com/itnewscloud/image/upload/v1597912961/No-image-available_dljkwb.png";
+    private final ImgNewsRepository imgNewsRepository;
+    private final Cloudinary cloudinary;
+
+
+    public  CloudinaryService(ImgNewsRepository imgNewsRepository){
+        this.imgNewsRepository = imgNewsRepository;
+        Map<String, String> valueMap = new HashMap<>();
+        valueMap.put("cloud_name", "itnewscloud");
+        valueMap.put("api_key", "442436395883333");
```

Comment:
secret credentials in code. hmm - is it good idea?

Reply:
refactoring

    @Value("${name}")
    private String cloudName;

    @Value("${key}")
    private String apiKey;

    @Value("${secret}")
    private String apiSecret;

    public  CloudinaryService(ImgNewsRepository imgNewsRepository){
        this.imgNewsRepository = imgNewsRepository;
        Map<String, String> valueMap = new HashMap<>();
        valueMap.put("cloud_name", cloudName);
        valueMap.put("api_key", apiKey);
        valueMap.put("api_secret", apiSecret);
        cloudinary = new Cloudinary(valueMap);
    }